spec/lib/importers/youtube_data_spec.rb

```diff
@@ -1,70 +1,207 @@
-require 'spec_helper'
+# frozen_string_literal: true
 
 describe YoutubeData do
   fixtures :affiliates, :rss_feeds, :rss_feed_urls, :youtube_profiles
 
   describe '.refresh' do
     let(:rss_feed) { mock_model RssFeed }
     let(:profile) { mock_model YoutubeProfile }
-    let(:youtube_data) { double YoutubeData }
+    let(:youtube_data) { double described_class }
+    let(:active_profiles) { double YoutubeProfile }
+    let(:refresh) do
+      t = Thread.new { described_class.refresh }
+      sleep 0.1
+      t.kill
```

Comment:
We can leave this for now, but one way to avoid this would be to set the `maximum_profile_updates_per_day` in the `test` environment to something extremely low, like 1 or 2. That way the loop would end quickly and automatically in specs, without the need to kill it. That could also simplify the specs, as we wouldn't have to stub that value to test the context `'when we have already updated the maximum number of profiles for today'`.

Reply:
An excellent idea, once we actually break the infinite loop, but as agreed, that's out of scope for this ticket.